File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        vector<int>v(2,-1);
        int low=0,high=nums.size()-1;
        int ans=-1;
        while(low<=high){
            int mid = low+(high-low)/2;
            if(nums[mid]==target){
                ans=mid;
                high=mid-1;
            }else if(nums[mid]>target){
                high=mid-1;
            }else{
                low=mid+1;
            }
        }
        if(ans!=-1){
            v[0]=ans;
            int l=ans,h=nums.size()-1;
            while(l<=h){
                int m=l+(h-l)/2;
                if(nums[m]==target){
                    ans=m;
                    l=m+1;
                }else{
                    h=m-1;
                }
            }
            v[1]=ans;
        }
        return v;
    }
};

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int left = solveLeft(nums,target);
        int right = solveRight(nums,target);
        return new int[]{left,right};
    }

    public int solveLeft(int []nums,int target) {
        int l=0,h=nums.length-1;
        int ans=-1;
        while(l<=h) {
            int mid = l+(h-l)/2;
            if(nums[mid] == target) {
                ans=mid;
                h=mid-1;
            }else if(nums[mid]<target) {
                l = mid+1;
            }else {
                h=mid-1;
            }
        }
        return ans;
    }

    public int solveRight(int []nums,int target) {
        int l=0,h=nums.length-1;
        int ans=-1;
        while(l<=h) {
            int mid = l+(h-l)/2;
            if(nums[mid] == target) {
                ans=mid;
                l=mid+1;
            }else if(nums[mid]<target) {
                l = mid+1;
            }else {
                h=mid-1;
            }
        }
        return ans;
    }
}